src/Reference/HasOne.php

```diff
@@ -225,10 +225,15 @@ public function ref($defaults = []): Model
             }
         }
 
+        // their model will be reloaded after saving our model to reflect changes in referenced fields
+        $theirModel->reload_after_save = false;
+
         $theirModel->onHook(Model::HOOK_AFTER_SAVE, function ($theirModel) {
-            $ourValue = $this->their_field ? $theirModel->get($this->their_field) : $theirModel->id;
+            $theirValue = $this->their_field ? $theirModel->get($this->their_field) : $theirModel->id;
+
+            $this->getOurField()->set($theirValue)->owner->save();
 
-            $this->getOurModel()->set($this->our_field, $ourValue);
+            $theirModel->reload();
```

Comment:
I think this is wrong.
Imagine we have `$user->hasOne(City::class)->addTitle()`, so it's `user.city_id = city.id` in SQL.

Now I change name of users city from Riga to London and save it.
What happens then? There is no need to save ourModel if theirField was not the one changed (dirty) in theirModel.

Why we need to reload city model (theirModel)? What maybe should be reloaded is users (ourModel) in case of some linked (title_field) or calculated fields.

Reply:
I made 2 changes here:
- delay the reloading of $theirModel to after setting $ourFieldValue to $theirFieldValue.
This is necesary as the `reload` cannot find the new city record (query on city is limited to 'only cities whose user.city_id = new city.id'. But there are no such users as the user.city_id has not been set to city.id yet. So we first set user.city_id=city.id and then reload city.
- save $ourModel when $ourFieldValue updated (original code was setting it but not saving). Now I added also check to only save when changed

Reply:
Ok, probably that's correct.